### ISSUE TITLE
Incorrect order of planning changes for concurrent records

### ISSUE THREAD

**Issue by hroost** [octodns/octodns#877]
Hi, thanks a lot for the octodns!

Recently I found an unexpected behavior when using octodns-sync to sync changes from the YAML provider to Gcore or Digitalocean (so doesn't depend on providers).

I'm using the following versions:
```
octodns==0.9.14
octodns-gcore==0.0.1
octodns-digitalocean==0.0.1
```

I get the issue when creating/changing records with concurrent types: for example A and CNAME.
Let's say we created A record for abc.example.com and then applied changes where we deleted A record and placed instead CNAME abc.example.com.
In some iterations octodns handle this properly (delete first then create) but sometimes (often when there are some other records with changes), I would say randomly, it's trying to first create CNAME and then delete A record - in this order we are getting exceptions from DNS provider.
I think the problem may be in the planning algorithm, it should always first process all deletes, and after that create the remaining changes.
However, this problem doesn't occur each time and seems a bit random.  

Here are a few catches that I was able to reproduce:

```
********************************************************************************
* example.com.
********************************************************************************
* gcore (GCoreProvider)
*   Delete <ARecord A 3600, mail.example.com., ['127.0.0.1']>
*   Create <CnameRecord CNAME 3600, mail.example.com., mail.yandex.ru.> (zones)
*   Summary: Creates=1, Updates=0, Deletes=1, Existing Records=56
********************************************************************************


2022-02-09T20:32:49  [140074006480704] INFO  GCoreProvider[gcore] apply: making 2 changes to example.com.
2022-02-09T20:32:49  [140074006480704] INFO  GCoreProvider[gcore] creating: Create <CnameRecord CNAME 3600, mail.example.com., mail.yandex.ru.> (zones)
Traceback (most recent call last):
  File "/root/octodns/venv/bin/octodns-sync", line 11, in <module>
    sys.exit(main())
  File "/root/octodns/venv/lib/python3.6/site-packages/octodns/cmds/sync.py", line 40, in main
    force=args.force)
  File "/root/octodns/venv/lib/python3.6/site-packages/octodns/manager.py", line 468, in sync
    total_changes += target.apply(plan)
  File "/root/octodns/venv/lib/python3.6/site-packages/octodns/provider/base.py", line 181, in apply
    self._apply(plan)
  File "/root/octodns/venv/lib/python3.6/site-packages/octodns_gcore/__init__.py", line 594, in _apply
    getattr(self, f"_apply_{class_name.lower()}")(change)
  File "/root/octodns/venv/lib/python3.6/site-packages/octodns_gcore/__init__.py", line 557, in _apply_create
    new.zone.name[:-1], new.fqdn, new._type, data
  File "/root/octodns/venv/lib/python3.6/site-packages/octodns_gcore/__init__.py", line 112, in record_create
    "POST", self._rrset_url(zone_name, rrset_name, type_), data=data
  File "/root/octodns/venv/lib/python3.6/site-packages/octodns_gcore/__init__.py", line 88, in _request
    r.raise_for_status()
  File "/root/octodns/venv/lib/python3.6/site-packages/requests/models.py", line 960, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 409 Client Error: Conflict for url: https://dnsapi.gcorelabs.com/v2/zones/example.com/mail.example.com./CNAME

```

```
********************************************************************************
* example.com.
********************************************************************************
* digitalocean (DigitalOceanProvider)
*   Create <ARecord A 300, abc.example.com., ['127.0.0.1']> (zones-test)
*   Delete <CnameRecord CNAME 300, abc.example.com., ololo.alala.ru.>
*   Summary: Creates=1, Updates=0, Deletes=1, Existing Records=15
********************************************************************************


2022-02-10T14:51:54  [140111425021760] INFO  DigitalOceanProvider[digitalocean] apply: making 2 changes to example.com.
Traceback (most recent call last):
  File "/root/octodns/venv/bin/octodns-sync", line 11, in <module>
    sys.exit(main())
  File "/root/octodns/venv/lib/python3.6/site-packages/octodns/cmds/sync.py", line 40, in main
    force=args.force)
  File "/root/octodns/venv/lib/python3.6/site-packages/octodns/manager.py", line 468, in sync
    total_changes += target.apply(plan)
  File "/root/octodns/venv/lib/python3.6/site-packages/octodns/provider/base.py", line 181, in apply
    self._apply(plan)
  File "/root/octodns/venv/lib/python3.6/site-packages/octodns_digitalocean/__init__.py", line 337, in _apply
    getattr(self, f'_apply_{class_name}')(change)
  File "/root/octodns/venv/lib/python3.6/site-packages/octodns_digitalocean/__init__.py", line 308, in _apply_Create
    self._client.record_create(new.zone.name[:-1], params)
  File "/root/octodns/venv/lib/python3.6/site-packages/octodns_digitalocean/__init__.py", line 102, in record_create
    self._request('POST', path, data=params)
  File "/root/octodns/venv/lib/python3.6/site-packages/octodns_digitalocean/__init__.py", line 47, in _request
    resp.raise_for_status()
  File "/root/octodns/venv/lib/python3.6/site-packages/requests/models.py", line 960, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 422 Client Error: Unprocessable Entity for url: https://api.digitalocean.com/v2/domains/example.com/records
```

**Comment by ross** [comment]
This is likely https://github.com/octodns/octodns/issues/656 which I thought had been addressed, but apparently not and stalebot recently closed it. I just reopened that issue. 

TL;DR is that some providers already order deletes before creates to solve this problem and that issue tracks exploring doing that in general. 

Closing this in favor of https://github.com/octodns/octodns/issues/656
